src/LOAuth.ts

```diff
@@ -112,6 +112,42 @@ class LOAuth {
     return btoa(JSON.stringify(state));
   }
 
+  /**
+   * This method looks for a cookie set by PHC-Login that allows the
+   * applications to exchange auth tokens over different subdomains
+   * see phc-login/src/util/handleAuthResponse
+   */
+  private _getDomainCookieAuthState() {
+    try {
+      if (!document.cookie) {
+        return;
+      }
+      const cookie = document.cookie
+        .split('; ')
+        .find((cookie) => cookie.startsWith(`${AUTH_STORAGE_KEY}=`));
+
+      const value = cookie && cookie.split('=')[1];
+      if (!value) {
+        return;
+      }
+
+      const storedData = JSON.parse(value);
+      this.token = this.client.createToken({
+        access_token: storedData.accessToken,
+        refresh_token: storedData.refreshToken,
+        token_type: 'Bearer',
+        expires: storedData.expiresAt
+      });
+      this.token.expiresIn(new Date(storedData.expiresAt - Date.now()));
+      this._storeTokenData(this.token);
+
+      // delete the cookie after it's read
+      document.cookie = `${AUTH_STORAGE_KEY}=;domain=.${storedData.cookieDomain};Max-Age=-9999;path=/;secure`;
+    } catch (err) {
+      return;
+    }
```

Comment:
Should we log, or callback somehow in the error case?

Reply:
Good catch, I didn't want to log to console directly and this lib doesn't have a logger object option. I figured since it'll just fallback to the normal flow after, I could silently ignore.